lib/transport/selenium-webdriver/browserstack/browserstack.js

```diff
@@ -44,6 +44,60 @@ class Browserstack extends AppiumBaseServer {
     });
   }
 
+  findBuildHashId(builds, buildName) {
+    const currentBuild = builds.find((item) => item.automation_build?.name === buildName);
+    if (currentBuild) {
+      return currentBuild.automation_build?.hashed_id;
+    }
+
+    return null; // Return null if no matching build is found
+  }
+
+  async fetchBuilds(url, offset) {
+    return await this.sendHttpRequest({
+      url: `${url}?status=running&limit=20&offset=${offset}`,
+      method: 'GET',
+      use_ssl: true,
+      port: 443,
+      auth: {
+        user: this.username,
+        pass: this.accessKey
+      }
+    });
+  }
+
+  async fetchBuildId(url, buildName) {
+    try {
+      let offset = 0;
+      let currentBuild;
+
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
+
+        const buildsResponse = await this.fetchBuilds(url, offset);
+
+        const builds = buildsResponse.builds || buildsResponse; // Handle different response structures
+        if (!builds || builds.length === 0) {
+          break;
+        }
+
+        const buildHashId = this.findBuildHashId(builds, buildName);
+        if (buildHashId) {
+
+          return buildHashId; // Return the hashed_id
+        }
+
+        offset += 20;
+      }
+
+      return null; // Return null if no matching build is found
```

Comment:
We shouldn't change the existing behaviour of the method. Right now, the method returns `undefined` if no hashed_id is found, we should continue doing that.